src/codeu/chat/server/Model.java

```diff
@@ -169,4 +228,178 @@ public void removeInterest(Uuid userId, Uuid interestId) {
+	  
+	  fw.write(gson.toJson(permissions, permissionType) + "\n");
+	  
+	  fw.write(gson.toJson(messages, messageType) + "\n");
+	  
+	  fw.write(gson.toJson(interestList, interestType) + "\n");	
+	  
+	  fw.write(gson.toJson(payloads, payloadType));
+	  
+	  fw.flush();
+	  fw.close();
+	  
+	  } catch (IOException e2) {
+		e2.printStackTrace();
+		System.out.println("Failed to save data to log!");
+	  }
+  }
+  
+  public boolean restore(File file) {
+	// check to see if there is a log to restore from 
+	if(!file.exists())
+	  return true;	
+	
+	BufferedReader br = null;
+	try {
+	  br = new BufferedReader(new FileReader(file.getAbsolutePath()));
+	} catch (FileNotFoundException e) {
+	  e.printStackTrace();
+	  return false;
+	}
+	
+	String completeText = "";
+	String line;
+	
+	// read the entirety of file into one String 
+	try {
+      while((line = br.readLine()) != null) {
+    	  completeText += line + "\n";
+      }
+      br.close();
+	} catch (IOException e) {
+	  e.printStackTrace();
+	  return false;
+	}
+	String[] jsonObjs = completeText.split("\n");
+	
+	if (jsonObjs.length == logSize) {
+	  // create gson object for deserializing
+	  GsonBuilder gb = new GsonBuilder();
+	  Type mapType = new TypeToken<Map<Uuid, UserType>>(){}.getType();
+	  gb.registerTypeAdapter(mapType, new PermissionAdaptor());
+	  Gson gson = gb.create();
+	  
+	  // convert from json to respective Store objects
+	  ArrayList<User> restoredUsers = gson.fromJson(jsonObjs[0].trim(), userType);
+	  ArrayList<ConversationHeader> restoredConvos = gson.fromJson(jsonObjs[1].trim(), conversationType);
+	  ArrayList<ConversationPermission> restoredPermissions = gson.fromJson(jsonObjs[2].trim(), permissionType);
+	  ArrayList<Message> restoredMessages = gson.fromJson(jsonObjs[3].trim(), messageType);
+	  ArrayList<Interest> restoredInterests = gson.fromJson(jsonObjs[4].trim(), interestType);
+	  ArrayList<ConversationPayload> restoredPayloads = gson.fromJson(jsonObjs[5].trim(), payloadType);
+	  
+	  // restore users
+	  for(int i = 0; i < restoredUsers.size(); i++) {
+	    add(restoredUsers.get(i));
+	  }
+	  
+	  // restore conversations
+	  for(int i = 0; i < restoredConvos.size(); i++) {
+	    add(restoredConvos.get(i));
+	  }
+	  
+	  // restore permissions
+	  for(int i = 0; i < restoredPermissions.size(); i++) {
+	    add(restoredPermissions.get(i));
+	  }
+	  
+	  // restore payloads
+	  for(int i = 0; i < restoredPayloads.size(); i++) {
+		add(restoredPayloads.get(i));
+	  }
+	  
+	  // restore messages
+	  for(int i = 0; i < restoredMessages.size(); i++) {
+	    add(restoredMessages.get(i));
+	  }
+	  
+	  // restore interests
+	  for(int i = 0; i < restoredInterests.size(); i++) {
+		Interest temp = restoredInterests.get(i);
+		addInterest(temp.id, temp.userId, temp.interestId, temp.type, temp.lastUpdate);
+	  }
+	  
+	  System.out.println("Sucessfully restored previous state of the Server");
+	  
+	  return true;
+	} else {
```

Comment:
For readability, I prefer 

    if (jsonObjs.length != logSize) {
        // print errors and return false
    }